File: Java/Classes/Donator.java
public class Donator extends User {
    private Offers offersList;

    public Donator(String name, String phone) {
        super(name, phone);
        this.offersList = new Offers(Organization.getInstance());
    }

    // In Donator.java
    public void add(RequestDonation donation) {
        try {
            offersList.add(donation, Organization.getInstance());
            System.out.println("Successfully added donation: " + donation.getEntity().getName() +
                    " - Quantity: " + donation.getQuantity()); // Add this debug line
        } catch (Exception e) {
            System.out.println("Error adding donation: " + e.getMessage());
        }
    }

    public void remove(RequestDonation donation) {
        offersList.remove(donation);
    }

    public void commit() {
        try {
            offersList.commit();
        } catch (Exception e) {
            System.out.println("Error committing offers: " + e.getMessage());
        }
    }

    public Offers getOffersList() {
        return offersList;
    }
}

File: C++/cpp_files/Donator.cpp
#include "Donator.h"
#include "Offers.h" // This will be created later

Donator::Donator(const std::string& name, const std::string& phone)
    : User(name, phone), offersList(new Offers()) {
}

Donator::~Donator() {
    delete offersList;
}

void Donator::add(int entityId, double quantity) {
    offersList->add(entityId, quantity);
}

void Donator::remove(int entityId) {
    offersList->remove(entityId);
}

void Donator::commit() {
    offersList->commit();
}

void Donator::monitor() const {
    offersList->monitor();
}

void Donator::reset() {
    offersList->reset();
}